src/pages/version/taskDuration/TaskDurationTable.tsx

```diff
@@ -34,46 +33,44 @@ export const TaskDurationTable: React.FC<Props> = ({
   const { sendEvent } = useVersionAnalytics(versionId);
   const { currentStatuses: statusOptions } = useTaskStatuses({ versionId });
 
-  const [
-    {
-      [PatchTasksQueryParams.TaskName]: taskName = "",
-      [PatchTasksQueryParams.Statuses]: statuses = [],
-      [PatchTasksQueryParams.Variant]: variant = "",
-      [PatchTasksQueryParams.Duration]: duration = "",
-    },
-  ] = useQueryParams();
+  const [queryParams, setQueryParams] = useQueryParams();
 
-  const filters = useMemo(
-    () => [
-      { id: PatchTasksQueryParams.TaskName, value: taskName },
-      {
-        id: PatchTasksQueryParams.Statuses,
-        value: Array.isArray(statuses) ? statuses : [statuses],
-      },
-      { id: PatchTasksQueryParams.Variant, value: variant },
-    ],
-    [taskName, statuses, variant]
+  const { initialFilters, initialSort } = useMemo(
+    () => getInitialParams(queryParams),
+    [] // eslint-disable-line react-hooks/exhaustive-deps
   );
 
-  const sorting = useMemo(
-    () => [
-      ...(duration && [
-        {
-          id: PatchTasksQueryParams.Duration,
-          desc: duration === SortDirection.Desc,
-        },
-      ]),
-    ],
-    [duration]
-  );
+  const [filters, setFilters] = useState<ColumnFiltersState>(initialFilters);
+  const [sorting, setSorting] = useState<SortingState>(initialSort);
```

Comment:
This seems like we are just duplicating the state that's already represented in the URL.

Reply:
We should be able to just rely on the urlstate.

Reply:
Yeah, react-table's handling of sort/filter is kinda weird... I went with this approach because it allows us to use the `onColumnFilterChange`/`onSortingChange` callbacks; using non "manual" filtering (to use react-table terminology) means that those functions are never called. But I changed approaches to have a `useEffect` hook update the query params instead.

Let me know what you think! If you like it I'll make a similar change to HostsTable.